### PR TITLE
Fix TeamCity env variables parser error #346

### PR DESCRIPTION
**Pull request by ghost**
I confirm that the pull-request:

- [X] Follows the contribution guidelines
- [X] Is based on my own work
- [X] Is in compliance with my employer

### CODE REVIEW

File: source/Nuke.Common/Utilities/String.IndexOf.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Nuke.Common.Utilities
+{
+    public static partial class StringExtensions
+    {
+        [Pure]
+        public static int IndexOfRegex(this string text, string expression)
+        {
+            var regex = new Regex(expression, RegexOptions.Compiled);
+            return regex.Match(text).Index + 1;
```
Comment (matkoch) [inline source/Nuke.Common/Utilities/String.IndexOf.cs]: I'm curious why `index + 1` ?

Comment (ghost) [reply source/Nuke.Common/Utilities/String.IndexOf.cs]: So it really depends on the regular expression, however for the one that teamcity uses it actually matches the character before the last = sign, so the +1 moves the index to the actual position of the equal sign. I think it's probably better if this method doesn't do the +1, and instead have the caller worry about such things. I will make a small update to the PR.

Comment (matkoch) [reply source/Nuke.Common/Utilities/String.IndexOf.cs]: That's what I thought. Thanks!

**Comment by matkoch** [comment]
Thanks for your contribution! 😄🎉

**Comment by matkoch** [comment]
I will release a 0.24.1 asap.

**Pull request merged**